iot-hub/Samples/module/ModuleSample/readme.md

```diff
@@ -0,0 +1,135 @@
+# Module Sample
+
+This sample illustrates how a module client handles its connection status updates, while sending telemetry to IoT Hub, and sending and receiving twin property updates.
+
+### Initialize the client:
+
+```csharp
+// Connection string:
+// Get the module connection string from Azure IoT Portal, or using Azure CLI. 
+// Azure portal - 
+// Navigate to your IoT Hub. From the left pane, under "Explorers", click on "IoT devices".
+// Click and navigate to your device. Under "Module Identities", click on your module.
+// Copy the connection strings listed (primary and/or secondary).
+// Azure CLI - 
+//  az iot hub module-identity connection-string show --device-id --module-id <module_id> [--key-type {primary, secondary}]
+//  --key-type is optional. It defaults to "primary".
+//
+// Transport type:
+// The transport to use to communicate with the IoT Hub. Possible values include Mqtt,
+// Mqtt_WebSocket_Only, Mqtt_Tcp_Only, Amqp, Amqp_WebSocket_Only, Amqp_Tcp_only, and Http1.
+//
+// Pass them to the application using command line parameters (see Parameters.cs).
+
+string connectionString = "<connection_string>";
+TransportType transportType = TransportType.Mqtt;
+moduleClient = ModuleClient.CreateFromConnectionString(connectionString, transportType);
+```
+
+### Send telemetry:
+
+```csharp
+var temperature = 25;
+var humidity = 70;
+string messagePayload = $"{{\"temperature\":{temperature},\"humidity\":{humidity}}}";
+
+var eventMessage = new Message(Encoding.UTF8.GetBytes(messagePayload))
+{
+    MessageId = Guid.NewGuid().ToString(),
+    ContentEncoding = Encoding.UTF8.ToString(),
+    ContentType = "application/json",
+};
+
+await moduleClient.SendEventAsync(message);
```

Comment:
worth talking about which classes of exceptions one might expect, and what app behavior is required based on that. Meaning, you might retry, reinit your client, or quit.

Reply:
Good point, how about I add a link to our sample where we handle the most common cases?

Reply:
Elaborate?

Reply:
We touch on those in the section "Module reconnection" below. I feel that if we add those instructions here again, it might be repetitive. 
I don't necessarily want to list exceptions based on operation - i.e. you might see _these_ exceptions while sending telemetry - since the list might not be exhaustive (we'd don't have an operation: exception mapping yet).

Thoughts?